### PR TITLE
Spec attribution triggering

### PR DESCRIPTION
**Pull request by johnivdel**
This PR specifies the following:
 - fetch monkeypatch for handling attribution triggers and blocking attribution trigger redirects
 - algorithm for choosing with source to attribute a trigger to
 - generating a report for a source and trigger

This also reorganizes the algorithms section to be split by source/trigger algorithms.

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/conversion-measurement-api/pull/167.html" title="Last updated on Jul 8, 2021, 8:10 PM UTC (93ceac1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/conversion-measurement-api/167/e6c049b...93ceac1.html" title="Last updated on Jul 8, 2021, 8:10 PM UTC (93ceac1)">Diff</a>

### CODE REVIEW

**Comment by johnivdel** [comment]
@csharrison could you please take a look?

File: index.bs [hunk 1]
```diff
@@ -385,59 +412,83 @@ To <dfn>process an attribution source</dfn> given an [=attribution source=] |sou
...
+Given a [=request=] |request|:
 
-1. If |request|'s [=request/current url's=] [=url/path=] is not `.well-known/attribution-reporting/trigger-attribution`,
-    return.
-1. If |request|'s [=request/redirect count=] is less than 1, return.
+1. If |request|'s [=request/current url's=] [=url/path=] is not equal to « ".well-known","attribution-reporting","trigger-attribution" »,
+    return <strong>allowed</strong>.
+1. Let |environment| be |request|'s [=request/window=].
+1. If |environment| is not an [=environment settings object=], return <strong>allowed</strong>.
+1. If |environment|'s [=environment/top-level origin=] is not a [=potentially trustworthy origin=], return <strong>allowed</strong>.
```
Comment (csharrison) [inline index.bs:450]: I think we also need to make sure the request's initiator is potentially trustworthy (i.e. the frame's origin)

File: index.bs [hunk 2]
```diff
@@ -385,59 +412,83 @@ To <dfn>process an attribution source</dfn> given an [=attribution source=] |sou
...
 
 <dfn>Max trigger data value</dfn> is a vendor specific integer which controls the potential values of [=attribution report/trigger data=].
 
 Issue: Formalize how to parse the query similar to URLSearchParams.
 
-<h3 algorithm id="triggering-attribution">Triggering attribution</h3>
+<h3 dfn id="should-block-response">Should internalResponse to request be blocked as attribution trigger</h3>
 
-To <dfn>trigger attribution</dfn> from a [=request=] |request|, run the following steps:
+Given a [=request=] |request|:
```
Comment (csharrison) [inline index.bs:444]: I think we need to check Permissions Policy here

File: index.bs [hunk 3]
```diff
@@ -385,59 +412,83 @@ To <dfn>process an attribution source</dfn> given an [=attribution source=] |sou
...
-1. Let |trigger| be the result of running [=obtain an attribution trigger=] with
-    |request|'s [=request/current url=] and |request|'s client.
-
-    Note: the restriction to require a redirect is needed to ensure that the
-    request's origin is aware and in control of triggering attribution.
+    |previousUrl|'s [=url/origin=], return <strong>allowed</strong>.
 
-1. Issue: Need to spec how to store |trigger|.
+    Note: The restriction to require a redirect is necessary to ensure that the
+     request's origin is aware and in control of the conversion registration.
```
[the remark this answers is not in the document]
Comment (johnivdel) [reply index.bs]: Done

File: index.bs [hunk 4]
```diff
@@ -385,59 +412,83 @@ To <dfn>process an attribution source</dfn> given an [=attribution source=] |sou
...
+    in descending order, with |a| being less than |b| if any of the following are true:
+      * |a|'s [=attribution source/priority=] is less than |b|'s [=attribution source/priority=].
+      * |a|'s [=attribution source/priority=] is equal to |b|'s [=attribution source/priority=] and |a|'s 
+         [=attribution source/source time=] is less than |b|'s [=attribution source/source time=].
+1. Let |sourceToAttribute| be the first item in |matchingSources|.
+1. [=list/Remove=] |sourceToAttribute| from |matchingSources|.
+1. For each |item| of |matchingSources|:
+    1. [=list/Remove=] |item| from the [=attribution source cache=].
+1. Let |report| be the result of running [=obtain a report=] with |sourceToAttribute| and |trigger|.
+1. Add |report| to the [=attribution report cache=].
```
[the remark this answers is not in the document]
Comment (johnivdel) [reply index.bs:492]: Done

File: index.bs [hunk 5]
```diff
@@ -385,59 +412,83 @@ To <dfn>process an attribution source</dfn> given an [=attribution source=] |sou
...
+1. Increment |sourceToAttribute|'s [=attribution source/number of reports=] value by 1.
+1. If |source|'s [=attribution source/number of reports=] value is greater than or equal to the
```
[the remark this answers is not in the document]
Comment (johnivdel) [reply index.bs:494]: Yes I'll address these in a future PR.

File: index.bs (hunk 2 above)
Comment (johnivdel) [reply index.bs:448]: Added the permissions policy definition here and elsewhere.

File: index.bs (hunk 1 above)
Comment (johnivdel) [reply index.bs:455]: Done

**Review by csharrison — APPROVED** [review]
LGTM

**Pull request merged**